vmdb/app/models/miq_provision_microsoft_workflow.rb

```diff
@@ -0,0 +1,63 @@
+class MiqProvisionMicrosoftWorkflow < MiqProvisionInfraWorkflow
+  def self.default_dialog_file
+    'miq_provision_dialogs'
+  end
+
+  def self.default_pre_dialog_file
+    'miq_provision_dialogs_pre'
+  end
+
+  def self.encrypted_options_fields
+    super + [:sysprep_password, :sysprep_domain_password]
+  end
+  def dialog_name_from_automate(message = 'get_dialog_name')
+    super(message, {'platform' => 'microsoft'})
+  end
+
+  def allowed_provision_types(_options = {})
+    {
+      "microsoft" => "Microsoft"
+    }
+  end
+
+  def self.allowed_templates_vendor
+    'microsoft'
+  end
+
+  def update_field_visibility(_options = {})
+    super
+
+    if get_value(@values[:vm_dynamic_memory])
+      show_fields(:edit, [:vm_minimum_memory, :vm_maximum_memory])
+    end
+  end
+
+  def allowed_vlans(options = {})
+    if @allowed_vlan_cache.nil?
+      @vlan_options ||= options
+      vlans = {}
+      src = get_source_and_targets
+      return vlans if src.blank?
+
+      hosts = nil
+      unless @vlan_options[:vlans] == false
+        rails_logger('allowed_vlans', 0)
+        hosts = get_selected_hosts(src)
+        hosts.each { |h| h.switches.each { |s| vlans[s.name] = s.name } }
+
+        # Remove certain networks
+        vlans.delete_if { |_k, v| v.include?('Service Console') || v.include?('VMkernel') }
+        rails_logger('allowed_vlans', 1)
+      end
+
+      unless @vlan_options[:dvs] == false
```

Comment:
This condition handles VMware DVS networks which are currently not stored in the database, you do not need this special logic.